app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,35 @@
 class PostsController < ApplicationController
-  def index; end
+  before_action :current_user, only: [:create]
 
-  def show; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+  end
+
+  def show
+    @post = Post.find(params[:id])
+    @user = User.find(@post.users_id)
+    @comments = @post.comments
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    @post.users_id = current_user.id
```

Comment:
- [ ] <b>REQUIRED:</b> As I said, when you say `current_user.posts.new()` the user_id is already initialized, so you don't need to re-initialize it again as you did on line 21. so you can simply remove that line.

Reply:
@rukundoeric thank you for your review, I have made the required changes